streams/src/test/java/org/apache/kafka/streams/kstream/JoinWindowsTest.java

```diff
@@ -101,102 +93,100 @@ public void startTimeShouldNotBeAfterEnd() {
         }
     }
 
-    @Test
-    public void untilShouldSetGraceDuration() {
-        final JoinWindows windowSpec = JoinWindows.of(ofMillis(ANY_SIZE));
-        final long windowSize = windowSpec.size();
-        assertEquals(windowSize, windowSpec.grace(ofMillis(windowSize)).gracePeriodMs());
-    }
-
     @Test
     public void gracePeriodShouldEnforceBoundaries() {
-        JoinWindows.of(ofMillis(3L)).grace(ofMillis(0L));
+        JoinWindows.ofTimeDifferenceAndGrace(ofMillis(3L), ofMillis(0L));
 
         try {
-            JoinWindows.of(ofMillis(3L)).grace(ofMillis(-1L));
+            JoinWindows.ofTimeDifferenceAndGrace(ofMillis(3L), ofMillis(-1L));
             fail("should not accept negatives");
         } catch (final IllegalArgumentException e) {
             //expected
         }
     }
 
     @Test
-    public void oldAPIShouldSetDefaultGracePeriod() {
```

Comment:
Ditto here, we should retain some version of this test and any others that are specifically intending to test the behavior of the old API (until the deprecation period has elapsed and we can remove it)